### PR TITLE
feat: add `@stdlib/assert/is-nonpositive-finite`

### LINKED ISSUE #1343: [RFC]: add `@stdlib/assert/is-nonpositive-finite`
### Description

This RFC proposes adding the package `@stdlib/assert/is-nonpositive-finite`.

The package should be similar in structure to [`@stdlib/assert/is-nonpositive-number`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/assert/is-nonpositive-number) and should have a similar implementation except negative infinity should evaluate to `false`.

```javascript
function isNonPositiveFinite( value ) { ... }
```

Package: `@stdlib/assert/is-nonpositive-finite`
Alias: `isNonPositiveFinite`

### Related Issues

None.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by shubhexists**
Resolves #1343
## Description

> What is the purpose of this pull request?

This pull request:

-   Add support for checking Non Positive Finite Integers

## Related Issues

> Does this pull request have any related issues?

#1343 

This pull request:

-   resolves #
-   fixes #

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

Tests
Screenshot from 2024-02-27 04-37-32
Screenshot from 2024-02-27 04-37-29

Benchmarks
Screenshot from 2024-02-27 04-38-04

Examples
Screenshot from 2024-02-27 04-38-30
## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/assert/is-nonpositive-finite/README.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+<!-- eslint-disable no-new-wrappers -->
+
+```javascript
+var Number = require( '@stdlib/number/ctor' );
+var NINF = require('@stdlib/constants/float64/ninf');
+
+var bool = isNonPositiveFinite( -5.0 );
+// returns true
+
+bool = isNonPositiveFinite(NINF);
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/assert/is-nonpositive-finite/README.md]:
```suggestion
bool = isNonPositiveFinite( NINF );
```

File: lib/node_modules/@stdlib/assert/is-nonpositive-finite/README.md [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+```javascript
+var Number = require( '@stdlib/number/ctor' );
+var isNonPositiveFinite = require( '@stdlib/assert/is-nonpositive-finite' );
+var NINF = require('@stdlib/constants/float64/ninf');
+
+var bool = isNonPositiveFinite( -5.0 );
+// returns true
+
+bool = isNonPositiveFinite(NINF);
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/assert/is-nonpositive-finite/README.md]:
```suggestion
bool = isNonPositiveFinite( NINF );
```

File: lib/node_modules/@stdlib/assert/is-nonpositive-finite/examples/index.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+var NINF = require('@stdlib/constants/float64/ninf');
+var isNonPositiveFinite = require( './../lib' );
+
+console.log( isNonPositiveFinite( -2 ));
+// => true
+
+console.log( isNonPositiveFinite( -2.99 ));
+// => true
+
+console.log(isNonPositiveFinite(NINF));
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/assert/is-nonpositive-finite/examples/index.js]:
```suggestion
console.log( isNonPositiveFinite( NINF ) );
```

File: lib/node_modules/@stdlib/assert/is-nonpositive-finite/docs/types/test.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+}
+
+// The compiler throws an error if the isPrimitive method is provided an unsupported number of arguments...
+{
+	isNonPositiveFinite.isPrimitive(); // $ExpectError
+	isNonPositiveFinite.isPrimitive( -2, 123 ); // $ExpectError
+}
+
+
+// Attached to main export is an isPrimitive method which returns a boolean...
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/assert/is-nonpositive-finite/docs/types/test.ts]:
```suggestion
// Attached to main export is an isObject method which returns a boolean...
```

File: lib/node_modules/@stdlib/assert/is-nonpositive-finite/docs/types/test.ts [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+	isNonPositiveFinite.isPrimitive( new Number( -2 ) ); // $ExpectType boolean
+	isNonPositiveFinite.isPrimitive( -2 ); // $ExpectType boolean
+}
+
+// The compiler throws an error if the isPrimitive method is provided an unsupported number of arguments...
+{
+	isNonPositiveFinite.isPrimitive(); // $ExpectError
+	isNonPositiveFinite.isPrimitive( -2, 123 ); // $ExpectError
+}
+
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/assert/is-nonpositive-finite/docs/types/test.ts:48]:
```suggestion
```

File: lib/node_modules/@stdlib/assert/is-nonpositive-finite/test/test.js [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+// TESTS //
+
+tape( 'main export is a function', function test( t ) {
+	t.ok( true, __filename );
+	t.strictEqual( typeof isNonPositiveFinite, 'function', 'main export is a function' );
+	t.end();
+});
+
+tape( 'attached to the main export is a method to test for a primitive number having a nonpositive number value', function test( t ) {
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/assert/is-nonpositive-finite/test/test.js]:
```suggestion
tape( 'attached to the main export is a method to test for a primitive number having a nonpositive finite number value', function test( t ) {
```

File: lib/node_modules/@stdlib/assert/is-nonpositive-finite/test/test.js [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+	t.equal( typeof isNonPositiveFinite.isPrimitive, 'function', 'export is a function' );
+	t.end();
+});
+
+tape( 'attached to the main export is a method to test for a number object having a nonpositive number value', function test( t ) {
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/assert/is-nonpositive-finite/test/test.js]:
```suggestion
tape( 'attached to the main export is a method to test for a number object having a nonpositive finite number value', function test( t ) {
```

File: lib/node_modules/@stdlib/assert/is-nonpositive-finite/docs/types/index.d.ts [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+// TypeScript Version: 4.1
+
+/**
+* Interface defining `isNonPositiveFinite` with methods for testing for primitives and objects, respectively.
+*/
+
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/assert/is-nonpositive-finite/docs/types/index.d.ts]:
```suggestion
```

File: lib/node_modules/@stdlib/assert/is-nonpositive-finite/lib/index.js [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+/**
+* Test if a value is a ragged nested array.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/assert/is-nonpositive-finite/lib/index.js]:
```suggestion
* Test if a value is a nonpositive finite number.
```

File: lib/node_modules/@stdlib/assert/is-nonpositive-finite/lib/index.js [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+*
+* @module @stdlib/assert/is-nonpositive-finite
+*
+* @example
+* var isNonPositiveFinite = require( '@stdlib/assert/is-nonpositive-finite' );
+*
+* var bool = isNonPositiveFinite(-2);
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/assert/is-nonpositive-finite/lib/index.js]:
```suggestion
* var bool = isNonPositiveFinite( -2 );
```

File: lib/node_modules/@stdlib/assert/is-nonpositive-finite/lib/index.js [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+* // returns true
+*
+* bool = isNonPositiveFinite(3);
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/assert/is-nonpositive-finite/lib/index.js]:
```suggestion
* bool = isNonPositiveFinite( 3 );
```

File: lib/node_modules/@stdlib/assert/is-nonpositive-finite/lib/index.js [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+* // returns false
+*
+* bool = isNonPositiveFinite('beep');
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/assert/is-nonpositive-finite/lib/index.js]:
```suggestion
* bool = isNonPositiveFinite( 'beep' );
```

File: lib/node_modules/@stdlib/assert/is-nonpositive-finite/lib/main.js [hunk 13]
```diff
@@ -0,0 +1,57 @@
...
+
+var NINF = require('@stdlib/constants/float64/ninf');
+var isPrimitive = require( './primitive.js' );
+var isObject = require( './object.js' );
+
+
+// MAIN //
+
+/**
+* Tests if a value is a number object has a non positive finite value.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/assert/is-nonpositive-finite/lib/main.js]:
```suggestion
* Tests if a value is a nonpositive finite number.
```

File: lib/node_modules/@stdlib/assert/is-nonpositive-finite/lib/main.js [hunk 14]
```diff
@@ -0,0 +1,57 @@
...
+* @example
+* var bool = isNonPositiveFinite( -3.0 );
+* // returns true
+*
+* @example
+* var bool = isNonPositiveFinite( new Number( -3.0 ) );
+* // returns true
+*
+* @example
+* var bool = isNonPositiveFinite(NINF);
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/assert/is-nonpositive-finite/lib/main.js]:
```suggestion
* var bool = isNonPositiveFinite( -Infinity );
```

File: lib/node_modules/@stdlib/assert/is-nonpositive-finite/lib/object.js [hunk 15]
```diff
@@ -0,0 +1,51 @@
...
+// MODULES //
+
+var NINF = require('@stdlib/constants/float64/ninf');
+var isNumber = require( '@stdlib/assert/is-number' ).isObject;
+
+
+// MAIN //
+
+/**
+* Tests if a value is a number object has a non positive finite value.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/assert/is-nonpositive-finite/lib/object.js]:
```suggestion
* Tests if a value is a number object having a non positive finite value.
```

File: lib/node_modules/@stdlib/assert/is-nonpositive-finite/lib/object.js [hunk 16]
```diff
@@ -0,0 +1,51 @@
...
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var NINF = require('@stdlib/constants/float64/ninf');
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/assert/is-nonpositive-finite/lib/object.js]:
```suggestion
var NINF = require( '@stdlib/constants/float64/ninf' );
```

File: lib/node_modules/@stdlib/assert/is-nonpositive-finite/lib/primitive.js [hunk 17]
```diff
@@ -0,0 +1,50 @@
...
+// MODULES //
+
+var NINF = require('@stdlib/constants/float64/ninf');
+var isNumber = require( '@stdlib/assert/is-number' ).isPrimitive;
+
+
+// MAIN //
+
+/**
+* Tests if a value is a number primitive has a non positive finite value.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/assert/is-nonpositive-finite/lib/primitive.js]:
```suggestion
* Tests if a value is a number primitive having a non positive finite value.
```

File: lib/node_modules/@stdlib/assert/is-nonpositive-finite/lib/primitive.js [hunk 18]
```diff
@@ -0,0 +1,50 @@
...
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var NINF = require('@stdlib/constants/float64/ninf');
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/assert/is-nonpositive-finite/lib/primitive.js]:
```suggestion
var NINF = require( '@stdlib/constants/float64/ninf' );
```

File: lib/node_modules/@stdlib/assert/is-nonpositive-finite/lib/primitive.js [hunk 19]
```diff
@@ -0,0 +1,50 @@
...
+* @example
+* var bool = isNonPositiveFinite( -3.0 );
+* // returns true
+*
+* @example
+* var bool = isNonPositiveFinite( new Number( -3.0 ) );
+* // returns false
+*/
+function isNonPositiveFinite( value ) {
+	return (isNumber( value ) && value <=0 && value !== NINF);
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/assert/is-nonpositive-finite/lib/primitive.js]:
```suggestion
	return ( isNumber( value ) && value <=0 && value !== NINF );
```

**Review by Planeshifter — APPROVED** [review]
Thanks for your contribution! Fixed a bunch of small code style issues, but otherwise this should be ready to land. Will merge once CI has passed.

**Comment by shubhexists** [comment]
Thank you sir for the suggestions.

**Pull request merged**